### PR TITLE
Feature/country search and filters

### PR DESCRIPTION
**Pull request by arkadiuszpalka**

### CODE REVIEW

File: app/src/main/java/pl/valueadd/restcountries/presentation/main/countries/list/CountryListFragment.kt [hunk 1]
```diff
@@ -48,7 +73,42 @@ class CountryListFragment : BaseMVPFragment<CountryListView, CountryListPresente
...
     }
 
-    private fun setupView() {
+    override fun onQueryTextSubmit(query: String?): Boolean =
+        onQueryTextChange(query)
+
+    override fun onQueryTextChange(newText: String?): Boolean {
+        presenter.onQueryChanged(newText ?: EMPTY)
+
+        return false
```
Comment (PiotrBandurski) [inline app/src/main/java/pl/valueadd/restcountries/presentation/main/countries/list/CountryListFragment.kt]: regarding documentation this should return `true` https://developer.android.com/reference/android/widget/SearchView.OnQueryTextListener.html#onQueryTextChange(java.lang.String)

File: app/src/main/java/pl/valueadd/restcountries/presentation/main/countries/list/CountryListPresenter.kt [hunk 2]
```diff
@@ -84,4 +126,11 @@ class CountryListPresenter
 
         it.showError(message)
     }
+
+    internal class Filters {
```
Comment (PiotrBandurski) [inline app/src/main/java/pl/valueadd/restcountries/presentation/main/countries/list/CountryListPresenter.kt:130]: Maybe you can use `enum` or `sealed` ?

File: app/src/main/java/pl/valueadd/restcountries/presentation/main/root/RootFragment.kt [hunk 3]
```diff
@@ -87,5 +91,17 @@ class RootFragment : BaseMVPFragment<RootView, RootPresenter>(R.layout.root_frag
...
+
+        initializeSearchView()
+    }
+
+    private fun initializeSearchView() {
+        searchView.layoutParams = Toolbar.LayoutParams(Gravity.END)
+
+        val countryListFragment = rootFragments[RootNavigation.COUNTRIES]
+
+        if (countryListFragment is SearchView.OnQueryTextListener) {
```
Comment (PiotrBandurski) [inline app/src/main/java/pl/valueadd/restcountries/presentation/main/root/RootFragment.kt]: Please make it more reusable

**Review by PiotrBandurski — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/valueadd/restcountries/presentation/main/countries/list/CountryListPresenter.kt (hunk 2 above)
Comment (arkadiuszpalka) [reply app/src/main/java/pl/valueadd/restcountries/presentation/main/countries/list/CountryListPresenter.kt:130]: I don't see the need to use `sealed` or `enum`

**Review by PiotrBandurski — APPROVED** [review]

**Pull request merged**